src/biogeochem/CNCStateUpdate1Mod.F90

```diff
@@ -71,9 +71,9 @@ subroutine CStateUpdateDynPatch(bounds, num_soilc_with_inactive, filter_soilc_wi
              c = filter_soilc_with_inactive(fc)
              cs_soil%decomp_cpools_vr_col(c,j,i_met_lit) = cs_soil%decomp_cpools_vr_col(c,j,i_met_lit) + &
                   cf_veg%dwt_frootc_to_litr_met_c_col(c,j) * dt
-             cs_soil%decomp_cpools_vr_col(c,j,i_cel_lit) = cs_soil%decomp_cpools_vr_col(c,j,i_cel_lit) + &
+             cs_soil%decomp_cpools_vr_col(c,j,i_litr2) = cs_soil%decomp_cpools_vr_col(c,j,i_litr2) + &
                   cf_veg%dwt_frootc_to_litr_cel_c_col(c,j) * dt
-             cs_soil%decomp_cpools_vr_col(c,j,i_lig_lit) = cs_soil%decomp_cpools_vr_col(c,j,i_lig_lit) + &
+             cs_soil%decomp_cpools_vr_col(c,j,i_litr3) = cs_soil%decomp_cpools_vr_col(c,j,i_litr3) + &
                   cf_veg%dwt_frootc_to_litr_lig_c_col(c,j) * dt
```

Comment:
In loops like this, we need to refactor the `dwt_frootc_to_litr_*_c_col` variables from 2d to 3d in order to replace the 3 repeating lines (one per litter pool) with a do-loop. Then MIMICS with 2 litter pools and BGC with 3 litter pools will run through here without a problem.

There are 3 such loops in CNCStateUpdate1, two in CNCStateUpdate2, and one in CNCStateUpdate1 and the pattern repeats in the corresponding N modules. As such, this refactor will touch a fair number of variables, so I will wait for your feedback @ekluzek and @wwieder before I embark.

Reply:
Marking this comment as important to discuss in Monday's mtg.

Reply:
Decided at meeting:
i_litr_min and i_litr_max for the suggested do-loops determined in the Cascade codes.
But keep i_met_lit, i_cel_lit, i_lig_lit and make them local to CascadeBGC
and similarly make the i_*_lit specific in CascadeMIMICS and CascadeCN
Also make the i_soil_* and i_micr_* specific.

Reply:
Since, this doesn't need to be dynamic (changing during run-time) the way it's done now makes the most sense.